### PR TITLE
[community]: Native RAG Support in Prem AI langchain 

### PR DESCRIPTION
**Pull request by Anindyadeep**
This PR adds native RAG support in langchain premai package. The same has been added in the docs too.

### CODE REVIEW

**Comment by startime-h** [comment]
Preparing review...

**Comment by Anindyadeep** [comment]
> Preparing review...

Let me know if this is ready for merge, thanks :)

**Comment by Anindyadeep** [comment]
Hi, let me know if we can merge this one thanks 

cc: @eyurtsev @baskaryan

File: libs/community/langchain_community/chat_models/premai.py [hunk 1]
```diff
@@ -268,21 +251,34 @@ def _llm_type(self) -> str:
...
+            "presence_penalty",
+            "logit_bias",
+            "stop",
+            "seed",
+        ]
+        keys_to_remove = []
+
+        for key in kwargs:
+            if key in kwargs_to_ignore:
+                logger.warning(f"WARNING: Parameter {key} is not supported in kwargs.")
```
Comment (baskaryan) [inline libs/community/langchain_community/chat_models/premai.py:276]: can we make this warnings.warning? (logger.warning is for things user can't change, warnings.warning is for things user can change)

Comment (Anindyadeep) [reply libs/community/langchain_community/chat_models/premai.py:276]: oh I see, sure, let me do that

Comment (Anindyadeep) [reply libs/community/langchain_community/chat_models/premai.py]: Done

**Pull request merged**